src/MainWindow.vala

```diff
@@ -379,342 +370,339 @@ namespace Quilter {
+                
+                if (gsettings.get_string("track-type") == "words") {
+                    statusbar.update_wordcount ();
+                    gsettings.set_string("track-type", "words");
+                } else if (gsettings.get_string("track-type") == "lines") {
+                    statusbar.update_linecount ();
+                    gsettings.set_string("track-type", "lines");
+                } else if (gsettings.get_string("track-type") == "chars") {
+                    statusbar.update_charcount ();
+                    gsettings.set_string("track-type", "chars");
+                }
+                statusbar.update_readtimecount ();
+            }
+            
+            private void action_preferences () {
+                var dialog = new Widgets.Preferences (this);
+                dialog.set_modal (true);
+                dialog.transient_for = this;
+                dialog.show_all ();
+            }
+            
+            private void action_cheatsheet () {
+                var dialog = new Widgets.Cheatsheet (this);
+                dialog.set_modal (true);
+                dialog.transient_for = this;
+                dialog.show_all ();
+            }
+            
+            private void action_export_pdf () {
+                Services.ExportUtils.export_pdf ();
+            }
+            
+            private void action_export_html () {
+                Services.ExportUtils.export_html ();
+            }
+            
+            private void render_func () {
                 preview_view_content.update_html_view ();
-                edit_view_content.buffer.set_modified (false);
+                if (edit_view_content.buffer.get_modified () == true) {
+                    preview_view_content.update_html_view ();
+                    edit_view_content.buffer.set_modified (false);
+                }
             }
-        }
-
-        public void show_sidebar () {
-            var settings = AppSettings.get_default ();
-            sidebar.show_this = settings.sidebar;
-            sidebar.reveal_child = settings.sidebar;
-        }
-
-        public void show_statusbar () {
-            var settings = AppSettings.get_default ();
-            statusbar.reveal_child = settings.statusbar;
-        }
-
-        public void show_searchbar () {
-            var settings = AppSettings.get_default ();
-            searchbar.reveal_child = settings.searchbar;
-        }
-
-        private void update_title () {
-            unowned Widgets.SideBarBox? row = sidebar.get_selected_row ();
-            if (row != null) {
-                toolbar.set_subtitle (row.title);
-            } else {
-                toolbar.set_subtitle (_("No Documents Open"));
+            
+            public void show_sidebar () {
+                
+                sidebar.show_this = gsettings.get_boolean("sidebar");
+                sidebar.reveal_child = gsettings.get_boolean("sidebar");
```

Comment:
Instead of this and the stuff up at 199, why not just bind the setting to the `reveal_child` property?

Reply:
I'm going to do that in outside commits after this PR